maas-api/internal/models/kserve_llmisvc.go

```diff
@@ -35,6 +40,131 @@ func (m *Manager) ListAvailableLLMs(ctx context.Context) ([]Model, error) {
 	return llmInferenceServicesToModels(instanceLLMs)
 }
 
+// ListAvailableLLMsForUser lists LLMInferenceServices that the user has access to.
+// This method filters models based on user's permissions by delegating to the gateway's AuthPolicy.
+func (m *Manager) ListAvailableLLMsForUser(ctx context.Context, user *token.UserContext, saToken string) ([]Model, error) {
```

Comment:
`user` seems to be only used for logging, maybe we can let the caller decide what to log and remove this param here?